crypto/kubernetes/secrets/component.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secrets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	contribCrypto "github.com/dapr/components-contrib/crypto"
+	kubeclient "github.com/dapr/components-contrib/internal/authentication/kubernetes"
+	internals "github.com/dapr/kit/crypto"
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	requestTimeout              = 30 * time.Second
```

Comment:
should this be configurable?

Reply:
I don't think for K8s this needs to be configurable, since the communication should be within the cluster IMHO. In case of failures here, users can retry (e.g. with resiliency).